### PR TITLE
Better fragment data formatter

### PR DESCRIPTION
**Pull request by c-Rolland**
## Subject

I am targeting this branch, because i have a little problem with the data format passed to the DOM who display the fragment's name and type. The little change format better the data passed to the DOM.

### CODE REVIEW

File: src/Resources/views/FragmentAdmin/form.html.twig [hunk 1]
```diff
@@ -1,9 +1,10 @@
 {% block form %}
     {% set fragmentName = admin.fragmentServices[form.vars.data.type].name|trans({}, 'SonataArticleBundle') %}
     {% set fragmentLabel = 'fragment'|trans({}, 'SonataArticleBundle') %}
+    {% set formdata = {'name': fragmentLabel ~ ' ' ~ fragmentName|escape, 'type' : form.vars.data.backofficeTitle|trans({}, 'SonataArticleBundle') } %}
     <div data-fragment-form="{{ form.vars.id }}"
          data-form-tmp="true"
-         data-formdata='{ "name": "{{ fragmentLabel }} {{ fragmentName|escape }}", "type": "{{ form.vars.data.backofficeTitle|trans({}, 'SonataArticleBundle') }}" }'
+         data-formdata="{{ formdata|json_encode() }}"
```
Comment (greg0ire) [inline src/Resources/views/FragmentAdmin/form.html.twig]: If the extra `formdata` is for clarity, here is another way to get some clarity: use linebreaks.
```suggestion
         data-formdata="{{ {
             'name': fragmentLabel ~ ' ' ~ fragmentName|escape, 
             'type' : form.vars.data.backofficeTitle|trans({}, 'SonataArticleBundle') 
         }|json_encode() }}"
```

**Review by greg0ire — COMMENTED** [review]
I agree that the way you propose to do this looks better, but I would like to know what is this little problem you had exactly.

File: src/Resources/views/FragmentAdmin/form.html.twig [hunk 2]
```diff
@@ -1,9 +1,13 @@
 {% block form %}
     {% set fragmentName = admin.fragmentServices[form.vars.data.type].name|trans({}, 'SonataArticleBundle') %}
     {% set fragmentLabel = 'fragment'|trans({}, 'SonataArticleBundle') %}
+    {% set formdata = {'name': fragmentLabel ~ ' ' ~ fragmentName|escape, 'type' : form.vars.data.backofficeTitle|trans({}, 'SonataArticleBundle') } %}
```
Comment (ChristianGripp) [inline src/Resources/views/FragmentAdmin/form.html.twig]: You can remove this

**Review by core23 — CHANGES_REQUESTED** [review]

**Comment by c-Rolland** [comment]
@greg0ire when i have special characters or html tags in "type" property nothing appears .. 
Capture d’écran 2019-11-13 à 15 44 52

i console.log the data given for jquery and when the "type" contains special characters or html tags we receive a string instead of a object ^^

**Comment by c-Rolland** [comment]
I don't know  what's causing that but this change fix it ^^

**Review by core23 — APPROVED** [review]

**Comment by c-Rolland** [comment]
i don't understand why the coverage fail i just modify a twig template ..

**Comment by OskarStark** [comment]
> i don't understand why the coverage fail i just modify a twig template ..

We now have 3 code lines more, but no additional testing, so the percentage is lower. Don't worry :)

**Review by OskarStark — APPROVED** [review]

**Comment by c-Rolland** [comment]
cool thanks .. ^^ @SonataCI can you tag this pls ? we need this fix in prod ^^

**Comment by core23** [comment]
SonataCI is a bot, so... No

But @sonata-project/release-managers can :+1:

**Comment by c-Rolland** [comment]
my bad sorry XD

**Pull request merged**